packages/neon_framework/packages/neon_http_client/lib/src/neon_http_client.dart

```diff
@@ -101,7 +101,11 @@ final class NeonHttpClient with http.BaseClient {
           interceptedRequest = await interceptor.interceptRequest(
             request: interceptedRequest,
           );
-        } catch (_, stackTrace) {
+        } catch (error, stackTrace) {
+          if (error is http.ClientException) {
+            rethrow;
+          }
+
```

Comment:
This needs to be documented somewhere.
What about the abstract interceptor class?